Stripe + frontend bug fixes

BohdanSizomin opened the pull request:
Stripe implemented but email sending on successful/failed payment not implemented yet
Minor frontend bugs fixes

BohdanSizomin committed:
Decimal Fields deleted,
Some values in Cart and Order models rounded
Stripe(without webhooks) integrated

Nazar-kob committed:
fixed bug If no address picked, then we got KeyError(data not passed to view)

BohdanSizomin committed:
Stripe with webhooks now works properly
But Email distribution on successfull/failed payment not implemented

BohdanSizomin committed:
Merge remote-tracking branch 'origin/nazar/fix/bugs_trello' into bohdan/feature/stripe

BohdanSizomin committed:
Flake8 fixes

Nazar-kob committed:
fixed bug If no variant picked inside product detail view then "Add to cart" button does nothing

BohdanSizomin committed:
Merge remote-tracking branch 'origin/nazar/fix/bugs_trello' into bohdan/feature/stripe

Nazar-kob committed:
fixed filter by attributes in products

BohdanSizomin committed:
Merge remote-tracking branch 'origin/nazar/fix/bugs_trello' into bohdan/feature/stripe

maverick0395 committed:
[F]: facebook login button fix

maverick0395 committed:
[F]: facebook button fix

maverick0395 committed:
[F] facebook and google buttons fix

BohdanSizomin committed:
Merge branch 'yaroslav/bug_fix/facebook_button' into bohdan/feature/stripe

Nazar-kob committed:
Fix imports in project

BohdanSizomin committed:
- some database attirbutes are unique now
- dashboard_api app now uses config from main app

BohdanSizomin committed:
merge with origin/nazar/fix/bugs_trello

Nazar-kob committed:
added /upload to gitignore

Nazar-kob committed:
add new commands

Nazar-kob committed:
chaged UPLOAD_FOLDER

Nazar-kob committed:
added image to product

Nazar-kob committed:
created new api endpoint for delete image

BohdanSizomin committed:
- forms placeholders added
- dashboard search now case insensitive
- minor elasticsearch fixes

Nazar-kob committed:
Merge branch 'bohdan/feature/stripe' of github.com:Simple2B/shop into bohdan/feature/stripe

Nazar-kob committed:
implemented delete image in product edit

BohdanSizomin committed:
- Now we can upload multiple images to our product
- When creating product, category cannot be None no longer
- Now if upload folder does not exist , then it will be created
- Product image deletion api replaced in "dashboard" blueprint
- node_modules excluded from flake8

BohdanSizomin committed:
- ProductEdit - image validation on client side implemented

BohdanSizomin committed:
- now we cannot provide wrong data to datepicker in dashboard

BohdanSizomin committed:
- Some imports optimized
- now we are able to create products with image in dashboard
- product id generator implemented
- fixed error when filtering by on_sale caused an error
- _macros.html - None option retured
- fixed searching through the next
models:Categories,Sales,Vouchers,ProductAttributes,ProductTypes,ShippingMethods,Pages,SiteMenus,DashboardMenus

BohdanSizomin committed:
- ProductAttributes is now creating without issue
- Fixed error when displaying wrong model for list of Product Types

kipod commented on .env.sample:
```diff
@@ -33,3 +33,6 @@ ADMIN_PASSWD=admin
 STORE_EMAIL=<need_to_set>
+
+STRIPE_PUBLISHABLE_KEY=<need_to_set>
+STRIPE_SECRET_KEY=<need_to_set>
```
EOL in EOF

kipod commented on docker-compose.yml:
```diff
@@ -34,6 +34,7 @@ services:
         #   memory: 100m
     environment:
       discovery.type: single-node
+      ELASTIC_PASSWORD: $ELASTIC_PASSWORD
```
"${ELASTIC_PASSWORD}"

kipod commented on flaskshop/account/models.py:
```diff
@@ -1,12 +1,10 @@
 from flask_login import UserMixin
 from sqlalchemy.ext.hybrid import hybrid_property
-from sqlalchemy.types import Enum
+from operator import or_
```
```python
from sqlalchemy.func import or_
```

kipod commented on flaskshop/account/views.py:
```diff
@@ -17,6 +17,7 @@
 from flask_babel import lazy_gettext
 import requests
 
+
```
pls revert

kipod commented on flaskshop/app.py:
```diff
@@ -59,6 +60,10 @@ def create_app(config_object=Config):
+    connections.create_connection(
+        hosts=[config_object.ES_URI],
+        http_auth=("elastic", config_object.ELASTIC_PASSWORD),
+    )
```
leak reference???

kipod commented on flaskshop/commands.py:
```diff
@@ -151,6 +151,25 @@ def createdb():
+@click.command()
+@with_appcontext
+def createsuperuser():
+    create_generator = chain(
```
add DOC STRING

kipod commented on flaskshop/commands.py:
```diff
@@ -151,6 +151,25 @@ def createdb():
+@click.command()
+@with_appcontext
+def dropdb():
+    db.drop_all()
```
Doc String

kipod commented on flaskshop/dashboard/views/discount.py:
```diff
@@ -66,7 +70,12 @@ def vouchers_manage(id=None):
+    title = request.args.get("title", type=str)
+    if title:
+        query = query.filter(Sale.title.ilike(f"%{title}%"))
+    pagination = query.paginate(page, 10)
```
CONST as a minimum, better config.PAGE_XX_SIZE

kipod commented on flaskshop/dashboard/views/product.py:
```diff
@@ -49,6 +67,7 @@ def attributes_manage(id=None):
     if form.validate_on_submit():
         if not id:
             attr = ProductAttribute()
+        attr.id = int(random.randint(1, 100000))
```
pls implement as func with check if this val exists

kipod commented on flaskshop/dashboard/views/user.py:
```diff
@@ -17,8 +17,8 @@ def users():
             or_(
-                User.username.like("%" + search_word + "%"),
-                User.email.like("%" + search_word + "%"),
+                User.username.ilike("%" + search_word + "%"),
```
```python
User.username.ilike(f"%{search_word}%"),
```

kipod commented on flaskshop/dashboard_api/views.py:
```diff
@@ -19,7 +18,8 @@ def item_del(cls, id):
         item.delete()
     except Exception as e:
-        return ApiResult({"r": 1, "msg": str(e)})
+        print(e)
```
use log(ERROR...

kipod commented on flaskshop/order/views.py:
```diff
@@ -43,42 +50,97 @@ def create_payment(token, payment_method):
+        product_names.append(line.product_name)
+
+    if not order:
+        return abort(404)
```
add Log!!!

kipod commented on flaskshop/order/views.py:
```diff
@@ -43,42 +50,97 @@ def create_payment(token, payment_method):
+        line_items=[
+            {
+                "name": ",".join(map(str, product_names)),
+                # "images": ["https://example.com/t-shirt.png"],
```
pls remove

kipod commented on flaskshop/order/views.py:
```diff
@@ -118,10 +185,18 @@ def flaskshop_load_blueprints(app):
     bp.add_url_rule("/<string:token>", view_func=show)
-    bp.add_url_rule("/pay/<string:token>/alipay", view_func=ali_pay)
-    bp.add_url_rule("/alipay/notify", view_func=ali_notify, methods=["POST"])
+    # bp.add_url_rule("/pay/<string:token>/alipay", view_func=ali_pay)
```
pls remove commented code

kipod reviewed (commented):
Thanks!!!

BohdanSizomin committed:
- If the product has no variants, it can't be added to the cart

BohdanSizomin committed:
- Now main page search function works
- Typo in elastic search fixed
- "Please pay before 00:30, else the order will auto closed" - string deleted
- Now stripe checkout contains an image of the first item of current order
- Pagination items per page is dynamical value now and can be provided by user in .env
- .env-sample file updated with new values
- When Category not set while creating product,it will flash a corresponding message

Nazar-kob committed:
repaired elasticsearch

Nazar-kob committed:
delete useless code

BohdanSizomin committed:
work
